### DISCUSSION TITLE
Remove options selected on a repeater form

### DISCUSSION THREAD

**Discussion by JayAyAre** [filamentphp/filament#11285] [Help]
### Package

Form builder

### Package Version

v3.0.0

### How can we help you?

i need remove the selected modules from the select, that is inside of a repeater

the code is actually that doesnot works is:

        $modules = Module::query()->pluck('name', 'id')->toArray();
        return $form
            ->schema([
                Forms\Components\Repeater::make('Modules')
                    ->schema([
                        Forms\Components\Select::make('modules')
                           ->live()
                            ->required()
                            ->disableOptionsWhenSelectedInSiblingRepeaterItems()
                            ->default('')
                            ->options(function () use ($modules) {
                                return $modules;
                            }),
                        Forms\Components\TextInput::make('discount')
                            ->label('Discount in %')
                            ->live()
                            ->numeric()
                            ->minValue(0)
                            ->maxValue(100)
                            ->required()
                            ->default(0)
                    ])->columns(2)
selected'),
dont use disableOptionsWhenSelectedInSiblingRepeaterItems because this put the selected option on gray, i wanted that the element dissapear directly
![image](https://github.com/filamentphp/filament/assets/73116131/6b8d0567-22a6-4021-b541-cd7088861725)

**Comment by leandrocfe** [comment]
```php
Select::make('type')
    ->options([
        'basic' => 'Basic',
        'advanced' => 'Advanced',
        'premium' => 'Premium',
    ])
    ->live(),

Repeater::make('modules')
    ->schema([
        Select::make('module_type')
            ->options([
                'basic' => 'Basic',
                'advanced' => 'Advanced',
                'premium' => 'Premium',
            ])
            ->disableOptionWhen(fn (Get $get, string $value): bool => $get('../../type') === $value),
    ])
```

**Reply by JayAyAre** [reply]
What I want is that in the variable $modules, when I select a module in a repeater, that it is no longer available in the others, I want it not to be shown on the screen (as in the image), I do not want it to appear disabled @leandrocfe

**Reply by leandrocfe** [reply]
You can filter the options

```php
Repeater::make('modules')
    ->schema([
        Select::make('module_type')
            ->options(function (Get $get): array {

                $options = [
                    'basic' => 'Basic',
                    'advanced' => 'Advanced',
                    'premium' => 'Premium',
                ];

                $filtered = Arr::where($options, function (string|int $value, string|int $key) use ($get) {
                    return $key !== $get('../../type');
                });

                return $filtered;
            }),
    ]),
```

You could filter it in your query

**Reply by JayAyAre** [reply]
Could you look my code and try to adapt that to mine?, i only know that i get the ids from modules with

                                $selectedModules = collect($get('../../Modules'))->pluck('modules')->toArray() ?: [];

bc im doing                             ->options(function (Forms\Get $get) use ($modules): array {

                                $selectedModules = collect($get('../../Modules'))->pluck('modules')->toArray() ?: [];

                                if($selectedModules) {
                                    $avalaibleModules = array_diff($modules, Module::query()->whereIn('id',$selectedModules)->pluck('name')->toArray());
                                }
                                return $avalaibleModules;
                            })
                            
                           But with this code, i take an option, and instantly change to another, but on the label i stay seeign the selected option. idk is so weird
                           
                                  $modules = Module::query()->pluck('name', 'id')->toArray();
        return $form
            ->schema([
                Forms\Components\Section::make('Plan Details')
                    ->schema([
                        Forms\Components\Select::make('name')
                            ->required()
                            ->label('Plan selected')
                    ])
                    ,
                Forms\Components\Repeater::make('Modules')
                    ->schema([
                        Forms\Components\Select::make('modules')
                            ->live()
                            ->required()
                            ->options(function (Forms\Get $get) use ($modules): array {

                                $selectedModules = collect($get('../../Modules'))->pluck('modules')->toArray() ?: [];

                                if($selectedModules) {
                                    $avalaibleModules = array_diff($modules, Module::query()->whereIn('id',$selectedModules)->pluck('name')->toArray());
                                }
                                return $avalaibleModules;
                            })
                            ->getOptionLabelFromRecordUsing(fn (array $state) => $state['modules'] ? $modules[$state['modules']] : 'No module selected'),
                        Forms\Components\TextInput::make('discount')
                            ->label('Discount in %')
                            ->live()
                            ->numeric()
                            ->minValue(0)
                            ->maxValue(100)
                            ->required()
                            ->default(0)
                    ])->columns(2)
                    ->columnSpanFull()
                    ->cloneable()
                    ->collapsible()
                    ->itemLabel(fn (array $state) => $state['modules'] ? $modules[$state['modules']] : 'No module selected'),

**Reply by leandrocfe** [reply]
Ahh ok, you are trying to hide in the select inside the repeater. I guess you can't do it.
You should use `disableOptionsWhenSelectedInSiblingRepeaterItems()`
